### PR TITLE
Examine Blocks

### LINKED ISSUE #67983: Examining AI and borgs shows their name twice
## Round ID:
N/A happens on local servers too

## Reproduction:
As any mob, examine an AI or borg. It will print their name twice
borgexamine

### PR DESCRIPTION
**Pull request by 13spacemen**
Adds Examine Blocks as seen on Skyrat and other servers, but tweaked a bit to look better

LIGHT MODE:
13pxlight

DARK MODE:
13pxdark

- [x] Add examine blocks to more places
- [x] Remove initial and final `*---------*` since that was a separator but is no longer needed
- [x] Tweak the colors a bit
- [x] Fix any bugs

Fixes https://github.com/tgstation/tgstation/issues/67983

## Why?
Helps multi-text outputs stand out from the rest of the chat, makes it way more readable when you use the various analyzers or examine someone or something

:cl: 13spacemen, Gandalf2k15
add: Examining and other chat outputs now display in blocks to make them easier to see
fix: AI and borgs no longer have their name show twice upon being examined
/:cl:

### CODE REVIEW

**Comment by san7890** [comment]
Any screenshots on it in Dark Mode?

**Comment by ArcaneMusic** [comment]
VERY clean

**Comment by timothymtorres** [comment]
This is amazing.  +1

File: code/game/objects/items/devices/scanners/slime_scanner.dm [hunk 1]
```diff
@@ -51,4 +51,4 @@
 	if(T.effectmod)
 		to_render += "\n[span_notice("Core mutation in progress: [T.effectmod]")]\
 					  \n[span_notice("Progress in core mutation: [T.applied] / [SLIME_EXTRACT_CROSSING_REQUIRED]")]"
-	to_chat(user, to_render + "\n========================")
+	to_chat(user, examine_block(to_render + "\n========================"))
```
Comment (timothymtorres) [inline code/game/objects/items/devices/scanners/slime_scanner.dm:54]: I saw you remove the `======` from the other parts of the code but why not here as well?

Comment (13spacemen) [reply code/game/objects/items/devices/scanners/slime_scanner.dm:54]: workin on it

**Comment by Farquaar** [comment]
How did we play this game for so many years without this?

File: code/modules/mob/mob.dm [hunk 2]
```diff
@@ -533,7 +533,12 @@
 	else
 		result = examinify.examine(src) // if a tree is examined but no client is there to see it, did the tree ever really exist?
 
-	to_chat(src, "<span class='infoplain'>[result.Join("\n")]</span>")
+	if(result.len)
+		for(var/i = 1, i <= result.len, i++)
+			if(!findtext(result[i], "<hr>"))
+				result[i] += "\n"
+
+	to_chat(src, "<div class='examine_block'><span class='infoplain'>[result.Join()]</span></div>")
```
Comment (LemonInTheDark) [inline code/modules/mob/mob.dm:541]:
```suggestion
	for(var/i in 1 to length(result))
		if(!findtext(result[i], "<hr>"))
			result[i] += "\n"
	to_chat(src, "<div class='examine_block'><span class='infoplain'>[result.Join()]</span></div>")
```
for(var/i in x to y) is faster, nicer, and functionally equivalent if the list's size doesn't change
to remove the trailing new line, do length(result) - 1

**Comment by LemonInTheDark** [comment]
if this code was ported from skyrat, could you credit the authors whos work you're using?

**Comment by 13spacemen** [comment]
> if this code was ported from skyrat, could you credit the authors whos work you're using?

the PR is https://github.com/Skyrat-SS13/Skyrat-tg/pull/8153, i think the author is @Gandalf2k15 , added as a credit

**Comment by LemonInTheDark** [comment]
that seems off, hm

**Comment by Wallemations** [comment]
Think we could get a screenshot of it with different text sizes?

File: tgui/packages/tgui-panel/styles/goon/chat-dark.scss [hunk 3]
```diff
@@ -900,6 +900,13 @@ em {
   margin-left: 3em;
 }
 
+.examine_block {
+  background: #1b1c1e;
+  border: 1px solid #a4bad6;
+  margin: 7px;
+  padding: 8px 12px;
+}
+
```
Comment (stylemistake) [inline tgui/packages/tgui-panel/styles/goon/chat-dark.scss:909]: You should use `em` units to write these styles, one `em` is equal to whatever font size in px you have set in chat settings, but try to stick to "one unit of margin is 0.5em" convention that we have in tgui.

Suggestion:

```suggestion
.examine_block {
  background: #1b1c1e;
  border: 1px solid #a4bad6;
  margin: 0.5em;
  padding: 0.5em 0.75em;
}
```

File: tgui/packages/tgui-panel/styles/goon/chat-light.scss [hunk 4]
```diff
@@ -933,6 +933,13 @@ h1.alert, h2.alert {
   margin-left: 3em;
 }
 
+.examine_block {
+  background: #f2f7fa;
+  border: 1px solid #111a27;
+  margin: 7px;
+  padding: 8px 12px;
+}
+
```
Comment (stylemistake) [inline tgui/packages/tgui-panel/styles/goon/chat-light.scss:942]:
```suggestion
.examine_block {
  background: #f2f7fa;
  border: 1px solid #111a27;
  margin: 0.5em;
  padding: 0.5em 0.75em;
}
```

**Comment by TrashDoxx** [comment]
I love it ❤️

**Comment by 13spacemen** [comment]
> Think we could get a screenshot of it with different text sizes?
21px
21pxbox
10px
10pxbox
13px
13pxdark

margins and padding use em now so they scale well with text size

**Review by stylemistake — APPROVED** [review]

**Pull request merged**